src/factories/OreFactory.js

```diff
@@ -0,0 +1,38 @@
+import { goldOre, ironOre, bronzeOre, silverOre } from '../data/ores'
+import { randomChoice } from '../common/utils'
+
+const SPAWNABLE_ORES = [goldOre, ironOre, bronzeOre, silverOre]
+
+/**
+ * Resource factory used for spawning ores
+ * @constructor
+ */
+export default class OreFactory {
+  constructor() {
+    this.oreOptions = []
+    for (let o of SPAWNABLE_ORES) {
+      this.oreOptions.push({
+        ore: o,
+        weight: o.spawnChance,
+      })
+    }
+  }
+
+  /**
+   * Generate resources
+   * @returns {Array} an array of ore resources
+   */
+  generate() {
+    return [this.spawn()]
+  }
+
+  /*
```

Comment:
nit: Extra asterisk here.

Reply:
not a nit, a requirement actually! i actually just realized that yesterday hah. nice catch